Uninitialized variable error when compiling with gcc 11

**Describe the bug**
QMCPACK doesn't compile correctly with gcc-11 

**To Reproduce**
Steps to reproduce the behavior:
1. `develop`
2. `cmake -GNinja -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ -D QMC_MPI=0 ../qmcpack`
3. `ninja`

**Expected behavior**
gcc-11 compilation must work

**System:**
Ubuntu 20.10 with default gcc-11 docker container image from [here](https://hub.docker.com/layers/ubuntu/library/ubuntu/impish-20210722/images/sha256-993294f9f183ba17a61857e31f589ddcda0e8f9636fc12d8fd34922f24539c36?context=explore) on Ubuntu 20.04 host.

**Additional context**
Error related to initialization:
```
$ ninja
[4/77] Building CXX object src/Numerics/tests/CMakeFiles/test_numerics.dir/test_grid_functor.cpp.o
FAILED: src/Numerics/tests/CMakeFiles/test_numerics.dir/test_grid_functor.cpp.o 
/usr/bin/g++ -DADD_ -DCATCH_CONFIG_ENABLE_BENCHMARKING -DH5_USE_16_API -DHAVE_CONFIG_H -DHAVE_LIBBOOST -DHAVE_LIBHDF5 -DOPENMP_NO_COMPLEX -DUSE_REAL_STRUCT_FACTOR -Drestrict=__restrict__ -I/home/user/qmcpack/src -Isrc -I/home/user/qmcpack/external_codes/boost_multi -I/home/user/qmcpack/external_codes/catch -I/home/user/qmcpack/src/Utilities -I/home/user/qmcpack/src/Containers -I/home/user/qmcpack/src/Platforms -I/home/user/qmcpack/src/io -isystem /usr/include/hdf5/serial -isystem /usr/include/libxml2 -Werror -fopenmp -finline-limit=1000 -fstrict-aliasing -funroll-all-loops -Wno-deprecated -Wvla -Wcomment -Wmisleading-indentation -Wmaybe-uninitialized -Wuninitialized -Wreorder -Wno-unknown-pragmas -Wno-sign-compare -Wsuggest-override -ffast-math -march=native -O3 -DNDEBUG -std=c++17 -MD -MT src/Numerics/tests/CMakeFiles/test_numerics.dir/test_grid_functor.cpp.o -MF src/Numerics/tests/CMakeFiles/test_numerics.dir/test_grid_functor.cpp.o.d -o src/Numerics/tests/CMakeFiles/test_numerics.dir/test_grid_functor.cpp.o -c /home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp: In function 'void qmcplusplus::____C_A_T_C_H____T_E_S_T____0()':
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:30:22: error: 'grid.qmcplusplus::LinearGrid<double, qmcplusplus::Vector<double, std::allocator<double> > >::<unnamed>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::lower_bound' may be used uninitialized [-Werror=maybe-uninitialized]
   30 |   LinearGrid<double> grid;
      |                      ^~~~
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:30:22: error: 'grid.qmcplusplus::LinearGrid<double, qmcplusplus::Vector<double, std::allocator<double> > >::<unnamed>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::upper_bound' may be used uninitialized [-Werror=maybe-uninitialized]
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:30:22: error: 'grid.qmcplusplus::LinearGrid<double, qmcplusplus::Vector<double, std::allocator<double> > >::<unnamed>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::Delta' may be used uninitialized [-Werror=maybe-uninitialized]
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:30:22: error: 'grid.qmcplusplus::LinearGrid<double>::<anonymous>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::DeltaInv' may be used uninitialized [-Werror=maybe-uninitialized]
In file included from /home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:15:
/home/user/qmcpack/src/Numerics/OneDimGridBase.h:35:8: error: 'grid.qmcplusplus::LinearGrid<double>::<anonymous>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::GridTag' may be used uninitialized [-Werror=maybe-uninitialized]
   35 | struct OneDimGridBase
      |        ^~~~~~~~~~~~~~
/home/user/qmcpack/src/Numerics/tests/test_grid_functor.cpp:30:22: note: 'grid.qmcplusplus::LinearGrid<double>::<anonymous>.qmcplusplus::OneDimGridBase<double, qmcplusplus::Vector<double, std::allocator<double> > >::GridTag' was declared here
   30 |   LinearGrid<double> grid;
      |                      ^~~~
cc1plus: all warnings being treated as errors
[21/77] Building CXX object src/QMCWaveFunctions/tests/CMakeFiles/test_wavefunction_trialwf.dir/test_TrialWaveFunction_diamondC_2x1x1.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/src/Numerics/OneDimGridBase.h
+++ b/src/Numerics/OneDimGridBase.h
@@ -38,13 +38,13 @@ struct OneDimGridBase
   typedef CT Array_t;
 
 
-  int GridTag;
-  int num_points;
-  value_type lower_bound;
-  value_type upper_bound;
+  int GridTag            = 0;
+  int num_points         = 0;
+  value_type lower_bound = T();
+  value_type upper_bound = T();
   ///differential spacing of the grid
-  value_type Delta;
-  double DeltaInv;
+  value_type Delta = T();
+  double DeltaInv  = 0.;
 
   ///array to store the radial grid data
   Array_t X;

--- a/src/Numerics/OneDimGridBase.h
+++ b/src/Numerics/OneDimGridBase.h
@@ -38,19 +38,19 @@ struct OneDimGridBase
   typedef CT Array_t;
 
 
-  int GridTag            = 0;
-  int num_points         = 0;
-  value_type lower_bound = T();
-  value_type upper_bound = T();
+  int GridTag;
+  int num_points;
+  value_type lower_bound;
+  value_type upper_bound;
   ///differential spacing of the grid
-  value_type Delta = T();
-  double DeltaInv  = 0.;
+  value_type Delta;
+  double DeltaInv;
 
   ///array to store the radial grid data
   Array_t X;
 
 
-  inline OneDimGridBase() : num_points(0) {}
+  inline OneDimGridBase() : GridTag(0), num_points(0), lower_bound(0), upper_bound(0), Delta(0), DeltaInv(0.) {}
 
   virtual std::unique_ptr<OneDimGridBase<T, CT>> makeClone() const = 0;
 

